docs/language/functions/fill.md

```diff
@@ -16,8 +16,18 @@ present in the output type `t` but not in the input.
 Filled fields are added with a `null` value.  Filling is useful when
 you want to be sure that all fields in a schema are present in a record.
 
+The order of newly added fields relative to fields already present is
+undefined.  If maintaining relative order is important, consider applying the
+[`order`](./order.md) or [`shape`](./shape) function.
```

Comment:
I stuck with "undefined" since I don't know the code. If we think there's something more detailed we could say that's not too complicated I'd be happy to set more explicit expectations (e.g., if/when it can be expected to append the new fields).